doc/rst/tools/mason/mason.rst

```diff
@@ -159,202 +194,364 @@ To try out different values at runtime, pass the values for ``number`` to ``maso
+
+  mason external --setup
+
+This command will install Spack and set it up so that it can be used easily through Mason.
+
+Let's use ``openSSL`` as an example since we used it in the system example. ``mason external search openSSL``
+will search for the package and produce the following output::
+
+  $ mason external search openSSL
+  ==> 2 packages.
+  openssl  r-openssl
+
+Obviously there are two types of the package listed, so we need to figure out which one to use.
+To find out more about a package, use ``mason external info <package>`` as follows::
 
+  $ mason external info openssl
+  Package:   openssl
 
-First a Mason project is created using ``mason new masonHDF5`` and the ``HDF5.chpl`` library
-is placed within the ``src/`` folder of the project and renamed ``masonHDF5.chpl``.
-A few examples with ``.h5`` data are placed in the ``examples/`` directory so that
-we can check out our new Mason package in action. After these steps, the project directory
-hierarchy is as follows::
+  Description:
+  OpenSSL is an open source project that provides a robust, commercial-
+  grade, and full-featured toolkit for the Transport Layer Security (TLS)
+  and Secure Sockets Layer (SSL) protocols. It is also a general-purpose
+  cryptography library.
 
+  Homepage: http://www.openssl.org
 
-    masonHDF5/
-      Mason.toml
-      Mason.lock
-      src/
-        masonHDF5.chpl
-      test/
-      example/
-        data/
-          sample_data.h5
-        hdf5Example.chpl
+  Tags:
+    None
 
-However, at this point, we have the wrapped function calls to the ``HDF5`` library included in
-the project but not the ``HDF5`` actual library itself. To search for and install packages such
-as ``HDF5``, simply use ``mason external search <package>`` and ``mason external install <package>``.
-Mason uses the ``Spack`` package specification.
+  Preferred version:
+    1.0.2k    http://www.openssl.org/source/openssl-1.0.2k.tar.gz
+
+  Safe versions:
+    1.1.0e    http://www.openssl.org/source/openssl-1.1.0e.tar.gz
+    1.1.0d    http://www.openssl.org/source/openssl-1.1.0d.tar.gz
+    1.1.0c    http://www.openssl.org/source/openssl-1.1.0c.tar.gz
+    1.0.2k    http://www.openssl.org/source/openssl-1.0.2k.tar.gz
+    1.0.2j    http://www.openssl.org/source/openssl-1.0.2j.tar.gz
+
+  Variants:
+   None
+
+  Installation Phases:
+   install
+
+  Build Dependencies:
+   zlib
+
+  Link Dependencies:
+   zlib
+
+  Run Dependencies:
+   None
+
+  Virtual Packages:
+   None
+
+
+
+The correct package has been found, but not yet installed. Let's fix that.
+We know that we want the preferred version which is ``1.0.2k``.
+The command to install ``openssl`` version ``1.0.2k`` would be::
+
+  mason external install openssl
+
+Since the version was left out, version ``1.0.2k`` is used because Mason
+will always take the preferred version. This is a case where Spack's
+spec expression syntax to specify exactly which package is desired.
```

Comment:
> This is a case where Spack's
spec expression syntax to specify exactly which package is desired.

I don't understand what this is trying to say.